addons/web/controllers/main.py

```diff
@@ -720,7 +720,8 @@ def write_header(self):
         # Write main header
         for i, fieldname in enumerate(self.field_names):
             self.write(0, i, fieldname, self.header_style)
-        self.worksheet.set_column(0, i, 30) # around 220 pixels
+        if self.field_names:
+            self.worksheet.set_column(0, i, 30) # around 220 pixels
```

Comment:
```suggestion
        self.worksheet.set_column(0, len(self.field_names), 30) # around 220 pixels
```
Seems like a clearer and more reliable fix. It solves the original issue by not accessing `i` at all (so doesn't care if that's not set), even though it's a bit misleading (as it'll set one column past the actual fields, and it always sets the width of the first column).

Alternatively,
```suggestion
        self.worksheet.set_column(0, max(0, len(self.field_names) - 1), 30) # around 220 pixels
```
which is a bit longer but more correct.

Reply:
Indeed this is way better :)